lib/src/impl/realtime/channels.dart

```diff
@@ -68,7 +72,12 @@ class RealtimePlatformChannel extends PlatformObject implements spec.RealtimeCha
         if(data!=null) "data": data
       });
     } on PlatformException catch (pe) {
-      throw spec.AblyException(pe.code, pe.message, pe.details);
+      if (hasAuthCallback && pe.code == ErrorCodes.authCallbackFailure.toString()) {
```

Comment:
Instead of using `toString()` everywhere, `ErrorCodes.xxx` should all be strings instead of integers.

Reply:
in PlatformException, it will be a string but in `ErrorInfo` it is an integer. This codes are exposed as public and so is ErrorInfo. I think it is better to have them as int.

Reply:
Ok, I just wanted to be sure this is intentional.

Reply:
Ok, lets resolve here. The discussion is open in another place already.